src/components/Select/Select.vue

```diff
@@ -0,0 +1,56 @@
+<template>
+  <div>
+    <label v-if="label" :class="labelClasses">{{ label }}</label>
+    <select :disabled="disabled" :class="[selectClasses, underline? underlineClasses: '']" @change="handelChange">
+      <option disabled selected value="">{{ placeholder }}</option>
+      <option :value="option.value" v-for="(option, index) in options" :key="index">
+        {{ option.name }}
+      </option>
+    </select>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import type { PropType } from 'vue'
+import type { InputSize } from '@/components/Input/types'
+import type { optionsType } from '@/components/Select/types'
+import { toRefs } from 'vue'
+import { useSelectClasses } from '@/components/Select/composables/useSelectClasses'
+
+const props = defineProps({
+  value: {
+    default: '',
+  },
+  label: {
+    type: String,
+    default: '',
+  },
+  options: {
+    type: Array as PropType<optionsType[]>,
+    default: () => [],
+  },
+  placeholder: {
+    type: String,
+    default: 'Please select one',
+  },
+  disabled: {
+    type: Boolean,
+    default: false,
+  },
+  underline: {
+    type: Boolean,
+    default: false,
+  },
+  size: {
+    type: String as PropType<InputSize>,
+    default: 'md',
+  },
+})
+
+const emit = defineEmits(['update:value'])
```

Comment:
We don't need to use multiple v-model bindings; it would be better to use just a single v-model

Full code component
```vue
<template>
  <div>
    <label v-if="label" :class="labelClasses">{{ label }}</label>
    <select v-model="model" :disabled="disabled" :class="[selectClasses, underline? underlineClasses: '']">
      <option disabled selected value="">{{ placeholder }}</option>
      <option :value="option.value" v-for="(option, index) in options" :key="index">
        {{ option.name }}
      </option>
    </select>
  </div>
</template>

<script lang="ts" setup>
import type { InputSize } from '@/components/Input/types'
import type { OptionsType } from '@/components/Select/types'
import { toRefs, computed } from 'vue'
import { useSelectClasses } from '@/components/Select/composables/useSelectClasses'

interface InputProps {
  modelValue?: string;
  label?: string;
  options?: OptionsType[];
  placeholder?: string;
  disabled?: boolean;
  underline?: boolean;
  size?: InputSize;
}
const props = withDefaults(defineProps<InputProps>(), {
  modelValue: '',
  label: '',
  options: () => [],
  placeholder: 'Please select one',
  disabled: false,
  underline: false,
  size: 'md',
})
const emit = defineEmits(['update:modelValue'])

const model = computed({
  get() {
    return props.modelValue
  },
  set(val) {
    emit('update:modelValue', val)
  },
})


const { selectClasses, underlineClasses, labelClasses } = useSelectClasses(toRefs(props))
</script>
```